langchain/src/schema/runnable.ts

```diff
@@ -82,4 +106,401 @@ export abstract class Runnable<
+      this.toJSON(),
+      _coerceToDict(input, "input")
+    );
+    let nextStepInput = input;
+    try {
+      for (const step of [this.first, ...this.middle]) {
+        nextStepInput = await step.invoke(
+          nextStepInput,
+          this._patchConfig(options, runManager?.getChild())
+        );
+      }
+    } catch (e) {
+      await runManager?.handleChainError(e);
+      throw e;
+    }
+    let concatSupported = true;
+    let finalOutput;
+    try {
+      const iterator = await this.last._streamIterator(
+        nextStepInput,
+        this._patchConfig(options, runManager?.getChild())
+      );
+      for await (const chunk of iterator) {
+        yield chunk;
+        if (concatSupported) {
+          if (finalOutput === undefined) {
+            finalOutput = chunk;
+          } else {
+            try {
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              finalOutput = (finalOutput as any).concat(chunk);
+            } catch (e) {
+              finalOutput = undefined;
+              concatSupported = false;
+            }
+          }
+        }
+      }
+    } catch (e) {
+      await runManager?.handleChainError(e);
+      throw e;
+    }
+    await runManager?.handleChainEnd(_coerceToDict(finalOutput, "output"));
+  }
+
+  pipe<NewRunOutput>(
+    coerceable: RunnableLike<RunOutput, NewRunOutput>
+  ): RunnableSequence<RunInput, NewRunOutput> {
+    if (RunnableSequence.isInstance(coerceable)) {
+      return new RunnableSequence({
+        first: this.first,
+        middle: this.middle.concat([
+          this.last,
+          coerceable.first,
+          ...coerceable.middle,
+        ]),
+        last: coerceable.last,
+      });
+    } else {
+      return new RunnableSequence({
+        first: this.first,
+        middle: [...this.middle, this.last],
+        last: _coerceToRunnable(coerceable),
+      });
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  static isInstance(thing: any): thing is RunnableSequence {
+    return Array.isArray(thing.middle) && Runnable.isInstance(thing);
+  }
+
+  static fromRunnables<RunInput, RunOutput>([first, ...runnables]: [
+    RunnableLike<RunInput>,
+    ...RunnableLike[],
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    RunnableLike<any, RunOutput>
+  ]) {
+    return new RunnableSequence<RunInput, RunOutput>({
+      first: _coerceToRunnable(first),
+      middle: runnables.slice(0, -1).map(_coerceToRunnable),
+      last: _coerceToRunnable(runnables[runnables.length - 1]),
+    });
+  }
+}
+
+export class RunnableMap<RunInput> extends Runnable<
+  RunInput,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Record<string, any>
+> {
+  lc_namespace = ["schema", "runnable"];
+
+  lc_serializable = true;
+
+  protected steps: Record<string, Runnable<RunInput>>;
+
+  constructor(steps: Record<string, RunnableLike<RunInput>>) {
```

Comment:
This needs to be an object with single key `steps`. Otherwise serialisation won't match py

Reply:
Ah right, thank you!